<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN""http://www.w3.org/TR/REC-html40/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc on Tue Jun 10 12:28:47 EST 2003 -->
<TITLE>
Index (JOONEGAP)
</TITLE>
<LINK REL ="stylesheet" TYPE="text/css" HREF="stylesheet.css" TITLE="Style">
</HEAD>
<SCRIPT>
function asd()
{
parent.document.title="Index (JOONEGAP)";
}
</SCRIPT>
<BODY BGCOLOR="white" onload="asd();">

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_top"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-all.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<A HREF="#_A_">A</A> <A HREF="#_C_">C</A> <A HREF="#_E_">E</A> <A HREF="#_G_">G</A> <A HREF="#_H_">H</A> <A HREF="#_I_">I</A> <A HREF="#_L_">L</A> <A HREF="#_M_">M</A> <A HREF="#_N_">N</A> <A HREF="#_O_">O</A> <A HREF="#_P_">P</A> <A HREF="#_R_">R</A> <A HREF="#_S_">S</A> <A HREF="#_T_">T</A> <A HREF="#_W_">W</A> <A HREF="#_X_">X</A> <HR>
<A NAME="_A_"><!-- --></A><H2>
<B>A</B></H2>
<DL>
<DT><A HREF="org/joonegap/RankedSelector.html#add(Configuration, org.jgap.Chromosome)"><B>add(Configuration, Chromosome)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/RankedSelector.html">RankedSelector</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_C_"><!-- --></A><H2>
<B>C</B></H2>
<DL>
<DT><A HREF="org/joonegap/ChromosomeManager.html"><B>ChromosomeManager</B></A> - class org.joonegap.<A HREF="org/joonegap/ChromosomeManager.html">ChromosomeManager</A>.<DD>ChromosomeManager manages the...<DT><A HREF="org/joonegap/ChromosomeManager.html#ChromosomeManager(java.io.File, boolean, int, int, Gene[], boolean, Configuration, Logger)"><B>ChromosomeManager(File, boolean, int, int, Gene[], boolean, Configuration, Logger)</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/ChromosomeManager.html">ChromosomeManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#cleanup()"><B>cleanup()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html#clear()"><B>clear()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#createUninitializedNeuralNetwork()"><B>createUninitializedNeuralNetwork()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Create an uninitialized neural network of the structure you desire.
<DT><A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html#createUninitializedNeuralNetwork()"><B>createUninitializedNeuralNetwork()</B></A> - 
Method in class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html">NIMNeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html#createUninitializedNeuralNetwork()"><B>createUninitializedNeuralNetwork()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html">NoughtsAndCrossesNeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html#createUninitializedNeuralNetwork()"><B>createUninitializedNeuralNetwork()</B></A> - 
Method in class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html">XORNeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html#CROSS"><B>CROSS</B></A> - 
Static variable in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html">NoughtOrCross</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#cycle(java.lang.Object, double[])"><B>cycle(Object, double[])</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Cycle the neural network, returning a sorted List of MeaningStrengthPairs for
 all output nodes.
</DL>
<HR>
<A NAME="_E_"><!-- --></A><H2>
<B>E</B></H2>
<DL>
<DT><A HREF="org/joonegap/RankedSelector.html#empty()"><B>empty()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/RankedSelector.html">RankedSelector</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NaturallySelectFirstGenotype.html#evolve()"><B>evolve()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NaturallySelectFirstGenotype.html">NaturallySelectFirstGenotype</A>
<DD>Override evolve() to do the natural selection first, <I>then</I>
 the mutation.
</DL>
<HR>
<A NAME="_G_"><!-- --></A><H2>
<B>G</B></H2>
<DL>
<DT><A HREF="org/joonegap/GenesScore.html#gameOver(java.lang.Boolean)"><B>gameOver(Boolean)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>Score is incremented if game is won, <I>decremented</I> if
 game is lost.
<DT><A HREF="org/joonegap/NeuralNetManager.html#gameOver(java.lang.Object, java.lang.Boolean)"><B>gameOver(Object, Boolean)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/ChromosomeManager.html#gameOver(java.lang.String, java.lang.Boolean)"><B>gameOver(String, Boolean)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/ChromosomeManager.html">ChromosomeManager</A>
<DD>Increment/decrement the chromosome's score
<DT><A HREF="org/joonegap/GaussianMutationOperator.html"><B>GaussianMutationOperator</B></A> - class org.joonegap.<A HREF="org/joonegap/GaussianMutationOperator.html">GaussianMutationOperator</A>.<DD>This genetic operator performs Gaussian mutation across
 all genes in a given Chromosome.<DT><A HREF="org/joonegap/GaussianMutationOperator.html#GaussianMutationOperator()"><B>GaussianMutationOperator()</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/GaussianMutationOperator.html">GaussianMutationOperator</A>
<DD>Constructs a GaussianMutationOperator with a default
 deviation of 0.05.
<DT><A HREF="org/joonegap/GaussianMutationOperator.html#GaussianMutationOperator(double)"><B>GaussianMutationOperator(double)</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/GaussianMutationOperator.html">GaussianMutationOperator</A>
<DD>Constructs a GaussianMutationOperator with the
 given deviation.
<DT><A HREF="org/joonegap/GeneNeuralNetCodec.html"><B>GeneNeuralNetCodec</B></A> - class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.html">GeneNeuralNetCodec</A>.<DD>Serializes an array of JGAP <CODE>Gene</CODE>s to and
 from a JOONE <CODE>NeuralNet</CODE>.<DT><A HREF="org/joonegap/GeneNeuralNetCodec.NoSuchWeightException.html"><B>GeneNeuralNetCodec.NoSuchWeightException</B></A> - exception org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.NoSuchWeightException.html">GeneNeuralNetCodec.NoSuchWeightException</A>.<DD>&nbsp;<DT><A HREF="org/joonegap/GeneNeuralNetCodec.NoSuchWeightException.html#GeneNeuralNetCodec.NoSuchWeightException()"><B>GeneNeuralNetCodec.NoSuchWeightException()</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.NoSuchWeightException.html">GeneNeuralNetCodec.NoSuchWeightException</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GeneNeuralNetCodec.html#GeneNeuralNetCodec()"><B>GeneNeuralNetCodec()</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.html">GeneNeuralNetCodec</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GenesScore.html"><B>GenesScore</B></A> - class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>.<DD>This simple class is used to track the activity
 of a given gene.<DT><A HREF="org/joonegap/GenesScore.html#GenesScore(java.lang.String)"><B>GenesScore(String)</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GeneNeuralNetCodec.html#GenesToNeuralNet(Gene[], NeuralNet)"><B>GenesToNeuralNet(Gene[], NeuralNet)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.html">GeneNeuralNetCodec</A>
<DD>Use the given array of genes to initialize the given neural network.
<DT><A HREF="org/joonegap/WeightGene.html#getAllele()"><B>getAllele()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html#getBoardAsArray()"><B>getBoardAsArray()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#getCustomConfiguration()"><B>getCustomConfiguration()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>In the majority of cases, clients can leave JOONEGAP to create and initialize
 its own JGAP Configuration object.
<DT><A HREF="org/joonegap/GenesScore.html#getGamesPlayed()"><B>getGamesPlayed()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#getGenerations()"><B>getGenerations()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/ChromosomeManager.html#getGenerations()"><B>getGenerations()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/ChromosomeManager.html">ChromosomeManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/MeaningStrengthPair.html#getMeaning()"><B>getMeaning()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/MeaningStrengthPair.html">MeaningStrengthPair</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#getMeaningOfOutputNodeAt(int)"><B>getMeaningOfOutputNodeAt(int)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Get the meaningful value of the neural network node at the given
 index.
<DT><A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html#getMeaningOfOutputNodeAt(int)"><B>getMeaningOfOutputNodeAt(int)</B></A> - 
Method in class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html">NIMNeuralNetManager</A>
<DD>Output nodes are numbered such that:
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html#getMeaningOfOutputNodeAt(int)"><B>getMeaningOfOutputNodeAt(int)</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html">NoughtsAndCrossesNeuralNetManager</A>
<DD>Output nodes are numbered such that:
<DT><A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html#getMeaningOfOutputNodeAt(int)"><B>getMeaningOfOutputNodeAt(int)</B></A> - 
Method in class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html">XORNeuralNetManager</A>
<DD>Output nodes are numbered such that:
<DT><A HREF="org/joonegap/GenesScore.html#getName()"><B>getName()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#getName(java.lang.Object)"><B>getName(Object)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Get the name of the object with the given identity.
<DT><A HREF="org/joonegap/WeightGene.html#getPersistentRepresentation()"><B>getPersistentRepresentation()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GenesScore.html#getScore()"><B>getScore()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/MeaningStrengthPair.html#getStrength()"><B>getStrength()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/MeaningStrengthPair.html">MeaningStrengthPair</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GeneNeuralNetCodec.html#getWeight(Gene[], int)"><B>getWeight(Gene[], int)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.html">GeneNeuralNetCodec</A>
<DD>Return the weight (or bias) at the given index in the
 given gene array.
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html#getWinner()"><B>getWinner()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>
<DD>See if 'noughts' or 'crosses' have won the match
</DL>
<HR>
<A NAME="_H_"><!-- --></A><H2>
<B>H</B></H2>
<DL>
<DT><A HREF="org/joonegap/WeightGene.html#hashCode()"><B>hashCode()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GenesScore.html#hold()"><B>hold()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#hold(java.lang.Object)"><B>hold(Object)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Hold the neural network with the given identity.
<DT><A HREF="org/joonegap/ChromosomeManager.html#holdChromosome()"><B>holdChromosome()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/ChromosomeManager.html">ChromosomeManager</A>
<DD>Get, at random, the next chromosome competitor.
</DL>
<HR>
<A NAME="_I_"><!-- --></A><H2>
<B>I</B></H2>
<DL>
<DT><A HREF="org/joonegap/GenesScore.html#isInUse()"><B>isInUse()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html#isNought()"><B>isNought()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html">NoughtOrCross</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_L_"><!-- --></A><H2>
<B>L</B></H2>
<DL>
<DT><A HREF="org/joonegap/GenesScore.html#LOSING_SCORE"><B>LOSING_SCORE</B></A> - 
Static variable in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_M_"><!-- --></A><H2>
<B>M</B></H2>
<DL>
<DT><A HREF="org/joonegap/WeightGene.html#m_activeConfiguration"><B>m_activeConfiguration</B></A> - 
Variable in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>The current active configuration that is in use.
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html#m_bIsNought"><B>m_bIsNought</B></A> - 
Variable in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html">NoughtOrCross</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#m_value"><B>m_value</B></A> - 
Variable in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>References the internal double value (allele) of this Gene.
<DT><A HREF="org/joonegap/samples/nim/NIMTest.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMTest.html">NIMTest</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesTest.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesTest.html">NoughtsAndCrossesTest</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/xor/XORTest.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORTest.html">XORTest</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/MeaningStrengthPair.html"><B>MeaningStrengthPair</B></A> - class org.joonegap.<A HREF="org/joonegap/MeaningStrengthPair.html">MeaningStrengthPair</A>.<DD>A <CODE>MeaningStrengthPair</CODE> encapsulates
 a 'meaning' (as defined by <CODE>NeuralNetManager.getMeaningOfOutputNodeAt</CODE>)
 together with its strength of activation.<DT><A HREF="org/joonegap/MeaningStrengthPair.html#MeaningStrengthPair(java.lang.Object, double)"><B>MeaningStrengthPair(Object, double)</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/MeaningStrengthPair.html">MeaningStrengthPair</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_N_"><!-- --></A><H2>
<B>N</B></H2>
<DL>
<DT><A HREF="org/joonegap/NaturallySelectFirstGenotype.html"><B>NaturallySelectFirstGenotype</B></A> - class org.joonegap.<A HREF="org/joonegap/NaturallySelectFirstGenotype.html">NaturallySelectFirstGenotype</A>.<DD>NaturallySelectFirstGenotype extends Genotype to override
 its <CODE>evolve</CODE> method, such that evolution takes
 place by doing natural selection <I>first</I>, <I>then</I>
 evolving the genotype.<DT><A HREF="org/joonegap/NaturallySelectFirstGenotype.html#NaturallySelectFirstGenotype(Configuration, org.jgap.Chromosome[])"><B>NaturallySelectFirstGenotype(Configuration, Chromosome[])</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/NaturallySelectFirstGenotype.html">NaturallySelectFirstGenotype</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html"><B>NeuralNetManager</B></A> - class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>.<DD>NeuralNetManager manages the...<DT><A HREF="org/joonegap/NeuralNetManager.html#NeuralNetManager(java.io.File, boolean, int, int)"><B>NeuralNetManager(File, boolean, int, int)</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Constructor (protected, to avoid accidental instantiation).
<DT><A HREF="org/joonegap/NeuralNetManager.html#NeuralNetManager(java.io.File, boolean, int, int, double, double, int, double[][], double[][])"><B>NeuralNetManager(File, boolean, int, int, double, double, int, double[][], double[][])</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>Constructor (protected, to avoid accidental instantiation).
<DT><A HREF="org/joonegap/GeneNeuralNetCodec.html#NeuralNetToGenes(NeuralNet)"><B>NeuralNetToGenes(NeuralNet)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.html">GeneNeuralNetCodec</A>
<DD>Convert the given neural network to an array of genes.
<DT><A HREF="org/joonegap/WeightGene.html#newGene(Configuration)"><B>newGene(Configuration)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html"><B>NIMNeuralNetManager</B></A> - class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html">NIMNeuralNetManager</A>.<DD>A neural network manager for playing NIM.<DT><A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html#NIMNeuralNetManager()"><B>NIMNeuralNetManager()</B></A> - 
Constructor for class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMNeuralNetManager.html">NIMNeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/nim/NIMTest.html"><B>NIMTest</B></A> - class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMTest.html">NIMTest</A>.<DD>This test has a <CODE>main</CODE> method which runs
 until a solution has been found to the game of NIM.<DT><A HREF="org/joonegap/samples/nim/NIMTest.html#NIMTest()"><B>NIMTest()</B></A> - 
Constructor for class org.joonegap.samples.nim.<A HREF="org/joonegap/samples/nim/NIMTest.html">NIMTest</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html#NOUGHT"><B>NOUGHT</B></A> - 
Static variable in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html">NoughtOrCross</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html"><B>NoughtOrCross</B></A> - class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html">NoughtOrCross</A>.<DD>&nbsp;<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html"><B>NoughtsAndCrossesGame</B></A> - class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>.<DD>&nbsp;<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html#NoughtsAndCrossesGame()"><B>NoughtsAndCrossesGame()</B></A> - 
Constructor for class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html"><B>NoughtsAndCrossesNeuralNetManager</B></A> - class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html">NoughtsAndCrossesNeuralNetManager</A>.<DD>A neural network manager for playing noughts and crosses.<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html#NoughtsAndCrossesNeuralNetManager()"><B>NoughtsAndCrossesNeuralNetManager()</B></A> - 
Constructor for class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesNeuralNetManager.html">NoughtsAndCrossesNeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesTest.html"><B>NoughtsAndCrossesTest</B></A> - class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesTest.html">NoughtsAndCrossesTest</A>.<DD>&nbsp;<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesTest.html#NoughtsAndCrossesTest()"><B>NoughtsAndCrossesTest()</B></A> - 
Constructor for class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesTest.html">NoughtsAndCrossesTest</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_O_"><!-- --></A><H2>
<B>O</B></H2>
<DL>
<DT><A HREF="org/joonegap/GaussianMutationOperator.html#operate(Configuration, org.jgap.Chromosome[], java.util.List)"><B>operate(Configuration, Chromosome[], List)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GaussianMutationOperator.html">GaussianMutationOperator</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/package-summary.html"><B>org.joonegap</B></A> - package org.joonegap<DD><DT><A HREF="org/joonegap/samples/nim/package-summary.html"><B>org.joonegap.samples.nim</B></A> - package org.joonegap.samples.nim<DD>&nbsp;<DT><A HREF="org/joonegap/samples/noughtsandcrosses/package-summary.html"><B>org.joonegap.samples.noughtsandcrosses</B></A> - package org.joonegap.samples.noughtsandcrosses<DD>&nbsp;<DT><A HREF="org/joonegap/samples/xor/package-summary.html"><B>org.joonegap.samples.xor</B></A> - package org.joonegap.samples.xor<DD>&nbsp;</DL>
<HR>
<A NAME="_P_"><!-- --></A><H2>
<B>P</B></H2>
<DL>
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html#play(org.joonegap.samples.noughtsandcrosses.NoughtOrCross, int, int)"><B>play(NoughtOrCross, int, int)</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>
<DD>Play the given type (nought or cross) at the given column and row.
</DL>
<HR>
<A NAME="_R_"><!-- --></A><H2>
<B>R</B></H2>
<DL>
<DT><A HREF="org/joonegap/RankedSelector.html"><B>RankedSelector</B></A> - class org.joonegap.<A HREF="org/joonegap/RankedSelector.html">RankedSelector</A>.<DD>RankedSelector is a JGAP NaturalSelector that selects the
 top 'n' (as given by the Genotype) ranked Chromosomes to go
 on to the next population.<DT><A HREF="org/joonegap/RankedSelector.html#RankedSelector()"><B>RankedSelector()</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/RankedSelector.html">RankedSelector</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GenesScore.html#release()"><B>release()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#release(java.lang.Object)"><B>release(Object)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/ChromosomeManager.html#releaseChromosome(java.lang.String)"><B>releaseChromosome(String)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/ChromosomeManager.html">ChromosomeManager</A>
<DD>Release the chromosome (so that it may be considered for evolution)
</DL>
<HR>
<A NAME="_S_"><!-- --></A><H2>
<B>S</B></H2>
<DL>
<DT><A HREF="org/joonegap/RankedSelector.html#select(Configuration, int)"><B>select(Configuration, int)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/RankedSelector.html">RankedSelector</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#setAllele(java.lang.Object)"><B>setAllele(Object)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/NeuralNetManager.html#setLogger(Logger)"><B>setLogger(Logger)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/NeuralNetManager.html">NeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#setToRandomValue(RandomGenerator)"><B>setToRandomValue(RandomGenerator)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/WeightGene.html#setValueFromPersistentRepresentation(java.lang.String)"><B>setValueFromPersistentRepresentation(String)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/GeneNeuralNetCodec.html#setWeight(double, java.util.List, int)"><B>setWeight(double, List, int)</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/GeneNeuralNetCodec.html">GeneNeuralNetCodec</A>
<DD>Store the given weight in the given list of genes at the given
 virtual index.
</DL>
<HR>
<A NAME="_T_"><!-- --></A><H2>
<B>T</B></H2>
<DL>
<DT><A HREF="org/joonegap/WeightGene.html#toString()"><B>toString()</B></A> - 
Method in class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html#toString()"><B>toString()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtsAndCrossesGame.html">NoughtsAndCrossesGame</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html#toString()"><B>toString()</B></A> - 
Method in class org.joonegap.samples.noughtsandcrosses.<A HREF="org/joonegap/samples/noughtsandcrosses/NoughtOrCross.html">NoughtOrCross</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_W_"><!-- --></A><H2>
<B>W</B></H2>
<DL>
<DT><A HREF="org/joonegap/WeightGene.html"><B>WeightGene</B></A> - class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>.<DD>This Gene is used to model individual weights (or bias') in the
 neural network.<DT><A HREF="org/joonegap/WeightGene.html#WeightGene()"><B>WeightGene()</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>Constructs a new DoubleGene with default settings.
<DT><A HREF="org/joonegap/WeightGene.html#WeightGene(Configuration)"><B>WeightGene(Configuration)</B></A> - 
Constructor for class org.joonegap.<A HREF="org/joonegap/WeightGene.html">WeightGene</A>
<DD>Constructs a new DoubleGene according to the given active
 configuration.
<DT><A HREF="org/joonegap/GenesScore.html#WINNING_SCORE"><B>WINNING_SCORE</B></A> - 
Static variable in class org.joonegap.<A HREF="org/joonegap/GenesScore.html">GenesScore</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_X_"><!-- --></A><H2>
<B>X</B></H2>
<DL>
<DT><A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html"><B>XORNeuralNetManager</B></A> - class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html">XORNeuralNetManager</A>.<DD>A neural network manager for figuring out the XOR problem.<DT><A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html#XORNeuralNetManager()"><B>XORNeuralNetManager()</B></A> - 
Constructor for class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORNeuralNetManager.html">XORNeuralNetManager</A>
<DD>&nbsp;
<DT><A HREF="org/joonegap/samples/xor/XORTest.html"><B>XORTest</B></A> - class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORTest.html">XORTest</A>.<DD>This test has a <CODE>main</CODE> method which runs
 until a solution has been found for the XOR problem.<DT><A HREF="org/joonegap/samples/xor/XORTest.html#XORTest()"><B>XORTest()</B></A> - 
Constructor for class org.joonegap.samples.xor.<A HREF="org/joonegap/samples/xor/XORTest.html">XORTest</A>
<DD>&nbsp;
</DL>
<HR>
<A HREF="#_A_">A</A> <A HREF="#_C_">C</A> <A HREF="#_E_">E</A> <A HREF="#_G_">G</A> <A HREF="#_H_">H</A> <A HREF="#_I_">I</A> <A HREF="#_L_">L</A> <A HREF="#_M_">M</A> <A HREF="#_N_">N</A> <A HREF="#_O_">O</A> <A HREF="#_P_">P</A> <A HREF="#_R_">R</A> <A HREF="#_S_">S</A> <A HREF="#_T_">T</A> <A HREF="#_W_">W</A> <A HREF="#_X_">X</A> 
<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_bottom"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-all.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>

</BODY>
</HTML>
