<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN""http://www.w3.org/TR/REC-html40/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc on Sat Sep 20 09:42:55 CDT 2003 -->
<TITLE>
BooleanGene (JGAP 1.0 API)
</TITLE>
<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">
</HEAD>
<SCRIPT>
function asd()
{
parent.document.title="BooleanGene (JGAP 1.0 API)";
}
</SCRIPT>
<BODY BGCOLOR="white" onload="asd();">

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_top"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="../../../org/jgap/impl/ChromosomePool.html"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="BooleanGene.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="../../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.jgap.impl</FONT>
<BR>
Class  BooleanGene</H2>
<PRE>
java.lang.Object
  |
  +--<B>org.jgap.impl.BooleanGene</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.lang.Comparable, <A HREF="../../../org/jgap/Gene.html">Gene</A>, java.io.Serializable</DD>
</DL>
<HR>
<DL>
<DT>public class <B>BooleanGene</B><DT>extends java.lang.Object<DT>implements <A HREF="../../../org/jgap/Gene.html">Gene</A>, java.io.Serializable</DL>

<P>
A Gene implementation that supports two possible values (alleles) for each
 gene: true and false.
 <p>
 NOTE: Since this Gene implementation only supports two different
 values (true and false), there's only a 50% chance that invocation
 of the setToRandomValue() method will actually change the value of
 this Gene (if it has a value). As a result, it may be desirable to
 use a higher overall mutation rate when this Gene implementation
 is in use.
<P>

<P>
<DL>
<DT><B>Since:</B><DD>1.0</DD>
</DD>
<DT><B>Author:</B><DD>Neil Rotstan</DD>
</DD>
<DT><B>See Also:</B><DD><A HREF="../../../serialized-form.html" TARGET="org.jgap.impl.BooleanGene">Serialized Form</A></DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->


<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Field Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;java.lang.Boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#FALSE_BOOLEAN">FALSE_BOOLEAN</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Shared constant representing the "false" boolean value.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.lang.Boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#m_value">m_value</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;References the internal boolean value of this Gene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;java.lang.Boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#TRUE_BOOLEAN">TRUE_BOOLEAN</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Shared constant representing the "true" boolean value.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#BooleanGene()">BooleanGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a new BooleanGene with default settings.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Method Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#booleanValue()">booleanValue</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the boolean value of this Gene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#cleanup()">cleanup</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Executed by the genetic engine when this Gene instance is no
 longer needed and should perform any necessary resource cleanup.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#compareTo(java.lang.Object)">compareTo</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this BooleanGene with the specified object for order.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this BooleanGene with the given object and returns true if
 the other object is a BooleanGene and has the same value as this
 BooleanGene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Object</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#getAllele()">getAllele</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the value represented by this Gene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#getPersistentRepresentation()">getPersistentRepresentation</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves a string representation of this Gene that includes any
 information required to reconstruct it at a later time, such as its
 value and internal state.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#hashCode()">hashCode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the hash code value of this BooleanGene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#newGene(org.jgap.Configuration)">newGene</A></B>(<A HREF="../../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Provides an implementation-independent means for creating new Gene
 instances.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#setAllele(java.lang.Object)">setAllele</A></B>(java.lang.Object&nbsp;a_newValue)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the value of this Gene to the new given value.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#setToRandomValue(org.jgap.RandomGenerator)">setToRandomValue</A></B>(<A HREF="../../../org/jgap/RandomGenerator.html">RandomGenerator</A>&nbsp;a_numberGenerator)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the value (allele) of this Gene to a random legal value.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#setValueFromPersistentRepresentation(java.lang.String)">setValueFromPersistentRepresentation</A></B>(java.lang.String&nbsp;a_representation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the value and internal state of this Gene from the string
 representation returned by a previous invocation of the
 getPersistentRepresentation() method.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgap/impl/BooleanGene.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves a string representation of this BooleanGene's value that
 may be useful for display purposes.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TD><B>Methods inherited from class java.lang.Object</B></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Field Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="TRUE_BOOLEAN"><!-- --></A><H3>
TRUE_BOOLEAN</H3>
<PRE>
protected static final java.lang.Boolean <B>TRUE_BOOLEAN</B></PRE>
<DL>
<DD>Shared constant representing the "true" boolean value. Shared constants
 are used to save memory so that a new Boolean object doesn't have to
 be constructed each time.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="FALSE_BOOLEAN"><!-- --></A><H3>
FALSE_BOOLEAN</H3>
<PRE>
protected static final java.lang.Boolean <B>FALSE_BOOLEAN</B></PRE>
<DL>
<DD>Shared constant representing the "false" boolean value. Shared constants
 are used to save memory so that a new Boolean object doesn't have to
 be constructed each time.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_value"><!-- --></A><H3>
m_value</H3>
<PRE>
protected java.lang.Boolean <B>m_value</B></PRE>
<DL>
<DD>References the internal boolean value of this Gene.
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="BooleanGene()"><!-- --></A><H3>
BooleanGene</H3>
<PRE>
public <B>BooleanGene</B>()</PRE>
<DL>
<DD>Constructs a new BooleanGene with default settings.
<P>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Method Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="newGene(org.jgap.Configuration)"><!-- --></A><H3>
newGene</H3>
<PRE>
public <A HREF="../../../org/jgap/Gene.html">Gene</A> <B>newGene</B>(<A HREF="../../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)</PRE>
<DL>
<DD>Provides an implementation-independent means for creating new Gene
 instances. The new instance that is created and returned should be
 setup with any implementation-dependent configuration that this Gene
 instance is setup with (aside from the actual value, of course). For
 example, if this Gene were setup with bounds on its value, then the
 Gene instance returned from this method should also be setup with
 those same bounds. This is important, as the JGAP core will invoke this
 method on each Gene in the sample Chromosome in order to create each
 new Gene in the same respective gene position for a new Chromosome.
 <p>
 It should be noted that nothing is guaranteed about the actual value
 of the returned Gene and it should therefore be considered to be
 undefined.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#newGene(org.jgap.Configuration)">newGene</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current active configuration.
<DT><B>Returns:</B><DD>A new Gene instance of the same type and with the same
         setup as this concrete Gene.</DL>
</DD>
</DL>
<HR>

<A NAME="setAllele(java.lang.Object)"><!-- --></A><H3>
setAllele</H3>
<PRE>
public void <B>setAllele</B>(java.lang.Object&nbsp;a_newValue)</PRE>
<DL>
<DD>Sets the value of this Gene to the new given value. This class
 expects the value to be a Boolean instance.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#setAllele(java.lang.Object)">setAllele</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_newValue</CODE> - the new value of this Gene instance.</DL>
</DD>
</DL>
<HR>

<A NAME="getPersistentRepresentation()"><!-- --></A><H3>
getPersistentRepresentation</H3>
<PRE>
public java.lang.String <B>getPersistentRepresentation</B>()
                                             throws java.lang.UnsupportedOperationException</PRE>
<DL>
<DD>Retrieves a string representation of this Gene that includes any
 information required to reconstruct it at a later time, such as its
 value and internal state. This string will be used to represent this
 Gene in XML persistence. This is an optional method but, if not
 implemented, XML persistence and possibly other features will not be
 available. An UnsupportedOperationException should be thrown if no
 implementation is provided.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#getPersistentRepresentation()">getPersistentRepresentation</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A string representation of this Gene's current state.
<DT><B>Throws:</B>
<DD><CODE>java.lang.UnsupportedOperationException</CODE> - to indicate that no implementation
         is provided for this method.</DL>
</DD>
</DL>
<HR>

<A NAME="setValueFromPersistentRepresentation(java.lang.String)"><!-- --></A><H3>
setValueFromPersistentRepresentation</H3>
<PRE>
public void <B>setValueFromPersistentRepresentation</B>(java.lang.String&nbsp;a_representation)
                                          throws <A HREF="../../../org/jgap/UnsupportedRepresentationException.html">UnsupportedRepresentationException</A></PRE>
<DL>
<DD>Sets the value and internal state of this Gene from the string
 representation returned by a previous invocation of the
 getPersistentRepresentation() method. This is an optional method but,
 if not implemented, XML persistence and possibly other features will not
 be available. An UnsupportedOperationException should be thrown if no
 implementation is provided.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#setValueFromPersistentRepresentation(java.lang.String)">setValueFromPersistentRepresentation</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_representation</CODE> - the string representation retrieved from a
                         prior call to the getPersistentRepresentation()
                         method.
<DT><B>Throws:</B>
<DD><CODE>java.lang.UnsupportedOperationException</CODE> - to indicate that no implementation
         is provided for this method.
<DD><CODE><A HREF="../../../org/jgap/UnsupportedRepresentationException.html">UnsupportedRepresentationException</A></CODE> - if this Gene implementation
         does not support the given string representation.</DL>
</DD>
</DL>
<HR>

<A NAME="getAllele()"><!-- --></A><H3>
getAllele</H3>
<PRE>
public java.lang.Object <B>getAllele</B>()</PRE>
<DL>
<DD>Retrieves the value represented by this Gene. All values returned
 by this class will be Boolean instances.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#getAllele()">getAllele</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the Boolean value of this Gene.</DL>
</DD>
</DL>
<HR>

<A NAME="booleanValue()"><!-- --></A><H3>
booleanValue</H3>
<PRE>
public boolean <B>booleanValue</B>()</PRE>
<DL>
<DD>Retrieves the boolean value of this Gene. This may be more convenient
 in some cases than the more general getAllele() method.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the boolean value of this Gene.</DL>
</DD>
</DL>
<HR>

<A NAME="setToRandomValue(org.jgap.RandomGenerator)"><!-- --></A><H3>
setToRandomValue</H3>
<PRE>
public void <B>setToRandomValue</B>(<A HREF="../../../org/jgap/RandomGenerator.html">RandomGenerator</A>&nbsp;a_numberGenerator)</PRE>
<DL>
<DD>Sets the value (allele) of this Gene to a random legal value. This
 method exists for the benefit of mutation and other operations that
 simply desire to randomize the value of a gene.
 <p>
 NOTE: Since this Gene implementation only supports two different
 values (true and false), there's only a 50% chance that invocation
 of this method will actually change the value of this Gene (if
 it has a value). As a result, it may be desirable to use a higher
 overall mutation rate when this Gene implementation is in use.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#setToRandomValue(org.jgap.RandomGenerator)">setToRandomValue</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_numberGenerator</CODE> - The random number generator that should be
                          used to create any random values. It's important
                          to use this generator to maintain the user's
                          flexibility to configure the genetic engine
                          to use the random number generator of their
                          choice.</DL>
</DD>
</DL>
<HR>

<A NAME="compareTo(java.lang.Object)"><!-- --></A><H3>
compareTo</H3>
<PRE>
public int <B>compareTo</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this BooleanGene with the specified object for order. A
 false value is considered to be less than a true value. A null value
 is considered to be less than any non-null value.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>compareTo</CODE> in interface <CODE>java.lang.Comparable</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - the BooleanGene to be compared.
<DT><B>Returns:</B><DD>a negative integer, zero, or a positive integer as this object
		is less than, equal to, or greater than the specified object.
<DT><B>Throws:</B>
<DD><CODE>java.lang.ClassCastException</CODE> - if the specified object's type prevents it
         from being compared to this BooleanGene.</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this BooleanGene with the given object and returns true if
 the other object is a BooleanGene and has the same value as this
 BooleanGene. Otherwise it returns false.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - the object to compare to this BooleanGene for equality.
<DT><B>Returns:</B><DD>true if this BooleanGene is equal to the given object,
         false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="hashCode()"><!-- --></A><H3>
hashCode</H3>
<PRE>
public int <B>hashCode</B>()</PRE>
<DL>
<DD>Retrieves the hash code value of this BooleanGene.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>hashCode</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>this BooleanGene's hash code.</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD>Retrieves a string representation of this BooleanGene's value that
 may be useful for display purposes.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>a string representation of this BooleanGene's value.</DL>
</DD>
</DL>
<HR>

<A NAME="cleanup()"><!-- --></A><H3>
cleanup</H3>
<PRE>
public void <B>cleanup</B>()</PRE>
<DL>
<DD>Executed by the genetic engine when this Gene instance is no
 longer needed and should perform any necessary resource cleanup.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../org/jgap/Gene.html#cleanup()">cleanup</A></CODE> in interface <CODE><A HREF="../../../org/jgap/Gene.html">Gene</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_bottom"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="../../../org/jgap/impl/ChromosomePool.html"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="BooleanGene.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="../../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>

</BODY>
</HTML>
