<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN""http://www.w3.org/TR/REC-html40/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc on Sat Sep 20 09:42:55 CDT 2003 -->
<TITLE>
Chromosome (JGAP 1.0 API)
</TITLE>
<LINK REL ="stylesheet" TYPE="text/css" HREF="../../stylesheet.css" TITLE="Style">
</HEAD>
<SCRIPT>
function asd()
{
parent.document.title="Chromosome (JGAP 1.0 API)";
}
</SCRIPT>
<BODY BGCOLOR="white" onload="asd();">

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_top"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgap/BulkFitnessFunction.html"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgap/Configuration.html"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Chromosome.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.jgap</FONT>
<BR>
Class  Chromosome</H2>
<PRE>
java.lang.Object
  |
  +--<B>org.jgap.Chromosome</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.lang.Cloneable, java.lang.Comparable, java.io.Serializable</DD>
</DL>
<HR>
<DL>
<DT>public class <B>Chromosome</B><DT>extends java.lang.Object<DT>implements java.lang.Comparable, java.lang.Cloneable, java.io.Serializable</DL>

<P>
Chromosomes represent potential solutions and consist of a fixed-length
 collection of genes. Each gene represents a discrete part of the solution.
 Each gene in the Chromosome may be backed by a different concrete
 implementation of the Gene interface, but all genes in a respective
 position (locus) must share the same concrete implementation across
 Chromosomes within a single population (genotype). In other words, gene 1
 in a chromosome must share the same concrete implementation as gene 1 in all
 other chromosomes in the population.
<P>

<P>
<DL>
<DT><B>Since:</B><DD>1.0</DD>
</DD>
<DT><B>Author:</B><DD>Neil Rotstan</DD>
</DD>
<DT><B>See Also:</B><DD><A HREF="../../serialized-form.html" TARGET="org.jgap.Chromosome">Serialized Form</A></DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->


<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Field Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgap/Configuration.html">Configuration</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#m_activeConfiguration">m_activeConfiguration</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The current active genetic configuration.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#m_fitnessValue">m_fitnessValue</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores the fitness value of this Chromosome as determined by the
 active fitness function.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgap/Gene.html">Gene</A>[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#m_genes">m_genes</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The array of Genes contained in this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#m_isSelectedForNextGeneration">m_isSelectedForNextGeneration</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Keeps track of whether or not this Chromosome has been selected by
 the natural selector to move on to the next generation.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#Chromosome(org.jgap.Configuration, org.jgap.Gene[])">Chromosome</A></B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration,
           <A HREF="../../org/jgap/Gene.html">Gene</A>[]&nbsp;a_initialGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs this Chromosome instance with the given array of gene values.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#Chromosome(org.jgap.Gene[])">Chromosome</A></B>(<A HREF="../../org/jgap/Gene.html">Gene</A>[]&nbsp;a_initialGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a Chromosome separate from any specific Configuration.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#Chromosome(org.jgap.Gene, int)">Chromosome</A></B>(<A HREF="../../org/jgap/Gene.html">Gene</A>&nbsp;a_sampleGene,
           int&nbsp;a_desiredSize)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a Chromosome of the given size separate from any specific
 Configuration.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Method Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#cleanup()">cleanup</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Invoked when this Chromosome is no longer needed and should perform
 any necessary cleanup.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Object</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#clone()">clone</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a copy of this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#compareTo(java.lang.Object)">compareTo</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares the given Chromosome to this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Chromosome against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#getFitnessValue()">getFitnessValue</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the fitness value of this Chromosome, as determined by the
 active fitness function.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgap/Gene.html">Gene</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#getGene(int)">getGene</A></B>(int&nbsp;a_desiredLocus)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the Gene at the given index (locus) within the Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgap/Gene.html">Gene</A>[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#getGenes()">getGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the set of genes that make up this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#hashCode()">hashCode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieve a hash code for this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#isSelectedForNextGeneration()">isSelectedForNextGeneration</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves whether this Chromosome has been selected by the natural
 selector to continue to the next generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../org/jgap/Chromosome.html">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#randomInitialChromosome(org.jgap.Configuration)">randomInitialChromosome</A></B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convenience method that returns a new Chromosome instance with its
 genes values (alleles) randomized.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#setActiveConfiguration(org.jgap.Configuration)">setActiveConfiguration</A></B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the active Configuration object on this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#setFitnessValue(int)">setFitnessValue</A></B>(int&nbsp;a_newFitnessValue)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the fitness value of this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#setIsSelectedForNextGeneration(boolean)">setIsSelectedForNextGeneration</A></B>(boolean&nbsp;a_isSelected)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets whether this Chromosome has been selected by the natural selector
 to continue to the next generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#size()">size</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the size of this Chromosome (the number of genes it contains).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Chromosome.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a string representation of this Chromosome, useful
 for some display purposes.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TD><B>Methods inherited from class java.lang.Object</B></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>finalize, getClass, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Field Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="m_activeConfiguration"><!-- --></A><H3>
m_activeConfiguration</H3>
<PRE>
protected transient <A HREF="../../org/jgap/Configuration.html">Configuration</A> <B>m_activeConfiguration</B></PRE>
<DL>
<DD>The current active genetic configuration.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_genes"><!-- --></A><H3>
m_genes</H3>
<PRE>
protected <A HREF="../../org/jgap/Gene.html">Gene</A>[] <B>m_genes</B></PRE>
<DL>
<DD>The array of Genes contained in this Chromosome.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_isSelectedForNextGeneration"><!-- --></A><H3>
m_isSelectedForNextGeneration</H3>
<PRE>
protected boolean <B>m_isSelectedForNextGeneration</B></PRE>
<DL>
<DD>Keeps track of whether or not this Chromosome has been selected by
 the natural selector to move on to the next generation.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_fitnessValue"><!-- --></A><H3>
m_fitnessValue</H3>
<PRE>
protected int <B>m_fitnessValue</B></PRE>
<DL>
<DD>Stores the fitness value of this Chromosome as determined by the
 active fitness function. A value of -1 indicates that this field
 has not yet been set with this Chromosome's fitness values (valid
 fitness values are always positive).
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="Chromosome(org.jgap.Gene, int)"><!-- --></A><H3>
Chromosome</H3>
<PRE>
public <B>Chromosome</B>(<A HREF="../../org/jgap/Gene.html">Gene</A>&nbsp;a_sampleGene,
                  int&nbsp;a_desiredSize)</PRE>
<DL>
<DD>Constructs a Chromosome of the given size separate from any specific
 Configuration. This constructor will use the given sample Gene to
 construct a new Chromosome instance containing genes all of the same
 type as the sample Gene. This can be useful for constructing sample
 chromosomes that use the same Gene type for all of their genes and that
 are to be used to setup a Configuration object.
<P>
<DT><B>Parameters:</B><DD><CODE>a_sampleGene</CODE> - A sample concrete Gene instance that will be used
                       as a template for all of the genes in this
                       Chromosome.<DD><CODE>a_desiredSize</CODE> - The desired size (number of genes) of this
                      Chromosome.</DL>
<HR>

<A NAME="Chromosome(org.jgap.Gene[])"><!-- --></A><H3>
Chromosome</H3>
<PRE>
public <B>Chromosome</B>(<A HREF="../../org/jgap/Gene.html">Gene</A>[]&nbsp;a_initialGenes)</PRE>
<DL>
<DD>Constructs a Chromosome separate from any specific Configuration. This
 can be useful for constructing sample chromosomes that are to be used
 to setup a Configuration object.
<P>
<DT><B>Parameters:</B><DD><CODE>a_initialGenes</CODE> - The genes of this Chromosome.</DL>
<HR>

<A NAME="Chromosome(org.jgap.Configuration, org.jgap.Gene[])"><!-- --></A><H3>
Chromosome</H3>
<PRE>
public <B>Chromosome</B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration,
                  <A HREF="../../org/jgap/Gene.html">Gene</A>[]&nbsp;a_initialGenes)
           throws <A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></PRE>
<DL>
<DD>Constructs this Chromosome instance with the given array of gene values.
 For convenience, the randomInitialChromosome method is provided
 that will take care of generating a Chromosome instance of a
 specified size. This constructor exists in case it's desirable
 to initialize Chromosomes with genes that contain specific values
 (alleles).
<P>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current, active Configuration object.<DD><CODE>a_initialGenes</CODE> - The array of genes that are to be contained
                       within this Chromosome instance.
<DT><B>Throws:</B>
<DD><CODE><A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></CODE> - if the given Configuration
         instance is null or invalid.
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if any of the given parameters are
         invalid, such as a null genes array or element.</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Method Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="setActiveConfiguration(org.jgap.Configuration)"><!-- --></A><H3>
setActiveConfiguration</H3>
<PRE>
public void <B>setActiveConfiguration</B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)
                            throws <A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></PRE>
<DL>
<DD>Sets the active Configuration object on this Chromosome. This method
 should be invoked immediately following deserialization of this
 Chromosome. If an active Configuration has already been set on this
 Chromosome, then this method will do nothing.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current active Configuration object
                              that is to be referenced internally by
                              this Chromosome instance.
<DT><B>Throws:</B>
<DD><CODE><A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></CODE> - if the Configuration object is
         null or cannot be locked because it is in an invalid or
         incomplete state.</DL>
</DD>
</DL>
<HR>

<A NAME="clone()"><!-- --></A><H3>
clone</H3>
<PRE>
public java.lang.Object <B>clone</B>()</PRE>
<DL>
<DD>Returns a copy of this Chromosome. The returned instance can evolve
 independently of this instance. Note that, if possible, this method
 will first attempt to acquire a Chromosome instance from the active
 ChromosomePool (if any) and set its value appropriately before
 returning it. If that is not possible, then a new Chromosome instance
 will be constructed and its value set appropriately before returning.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>clone</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A copy of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="getGene(int)"><!-- --></A><H3>
getGene</H3>
<PRE>
public <A HREF="../../org/jgap/Gene.html">Gene</A> <B>getGene</B>(int&nbsp;a_desiredLocus)</PRE>
<DL>
<DD>Returns the Gene at the given index (locus) within the Chromosome. The
 first gene is at index zero and the last gene is at the index equal to
 the size of this Chromosome - 1.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The Gene at the given index.</DL>
</DD>
</DL>
<HR>

<A NAME="getGenes()"><!-- --></A><H3>
getGenes</H3>
<PRE>
public <A HREF="../../org/jgap/Gene.html">Gene</A>[] <B>getGenes</B>()</PRE>
<DL>
<DD>Retrieves the set of genes that make up this Chromosome. This method
 exists primarily for the benefit of GeneticOperators that require the
 ability to manipulate Chromosomes at a low level.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>an array of the Genes contained within this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="size()"><!-- --></A><H3>
size</H3>
<PRE>
public int <B>size</B>()</PRE>
<DL>
<DD>Returns the size of this Chromosome (the number of genes it contains).
 A Chromosome's size is constant and will never change.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The number of genes contained within this Chromosome instance.</DL>
</DD>
</DL>
<HR>

<A NAME="getFitnessValue()"><!-- --></A><H3>
getFitnessValue</H3>
<PRE>
public int <B>getFitnessValue</B>()</PRE>
<DL>
<DD>Retrieves the fitness value of this Chromosome, as determined by the
 active fitness function. If a bulk fitness function is in use and
 has not yet assigned a fitness value to this Chromosome, then -1 is
 returned.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>a positive integer value representing the fitness of this
         Chromosome, or -1 if a bulk fitness function is in use and has
         not yet assigned a fitness value to this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="setFitnessValue(int)"><!-- --></A><H3>
setFitnessValue</H3>
<PRE>
public void <B>setFitnessValue</B>(int&nbsp;a_newFitnessValue)</PRE>
<DL>
<DD>Sets the fitness value of this Chromosome. This method is for use
 by bulk fitness functions and should not be invokved from anything
 else.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_newFitnessValue</CODE> - a positive integer representing the fitness
                          of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD>Returns a string representation of this Chromosome, useful
 for some display purposes.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A string representation of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="randomInitialChromosome(org.jgap.Configuration)"><!-- --></A><H3>
randomInitialChromosome</H3>
<PRE>
public static <A HREF="../../org/jgap/Chromosome.html">Chromosome</A> <B>randomInitialChromosome</B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)
                                          throws <A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></PRE>
<DL>
<DD>Convenience method that returns a new Chromosome instance with its
 genes values (alleles) randomized. Note that, if possible, this method
 will acquire a Chromosome instance from the active ChromosomePool
 (if any) and then randomize its gene values before returning it. If a
 Chromosome cannot be acquired from the pool, then a new instance will
 be constructed and its gene values randomized before returning it.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current active configuration.
<DT><B>Throws:</B>
<DD><CODE><A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></CODE> - if the given Configuration
         instance is invalid.
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if the given Configuration instance
         is null.</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this Chromosome against the specified object. The result is
 true if and the argument is an instance of the Chromosome class
 and has a set of genes equal to this one.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="hashCode()"><!-- --></A><H3>
hashCode</H3>
<PRE>
public int <B>hashCode</B>()</PRE>
<DL>
<DD>Retrieve a hash code for this Chromosome.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>hashCode</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the hash code of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="compareTo(java.lang.Object)"><!-- --></A><H3>
compareTo</H3>
<PRE>
public int <B>compareTo</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares the given Chromosome to this Chromosome. This chromosome is
 considered to be "less than" the given chromosome if it has a fewer
 number of genes or if any of its gene values (alleles) are less than
 their corresponding gene values in the other chromosome.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>compareTo</CODE> in interface <CODE>java.lang.Comparable</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The Chromosome against which to compare this chromosome.
<DT><B>Returns:</B><DD>a negative number if this chromosome is "less than" the given
         chromosome, zero if they are equal to each other, and a positive
         number if this chromosome is "greater than" the given chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="setIsSelectedForNextGeneration(boolean)"><!-- --></A><H3>
setIsSelectedForNextGeneration</H3>
<PRE>
public void <B>setIsSelectedForNextGeneration</B>(boolean&nbsp;a_isSelected)</PRE>
<DL>
<DD>Sets whether this Chromosome has been selected by the natural selector
 to continue to the next generation.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_isSelected</CODE> - true if this Chromosome has been selected, false
                     otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="isSelectedForNextGeneration()"><!-- --></A><H3>
isSelectedForNextGeneration</H3>
<PRE>
public boolean <B>isSelectedForNextGeneration</B>()</PRE>
<DL>
<DD>Retrieves whether this Chromosome has been selected by the natural
 selector to continue to the next generation.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>true if this Chromosome has been selected, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="cleanup()"><!-- --></A><H3>
cleanup</H3>
<PRE>
public void <B>cleanup</B>()</PRE>
<DL>
<DD>Invoked when this Chromosome is no longer needed and should perform
 any necessary cleanup. Note that this method will attempt to release
 this Chromosome instance to the active ChromosomePool, if any.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_bottom"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgap/BulkFitnessFunction.html"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgap/Configuration.html"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Chromosome.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>

</BODY>
</HTML>
