<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN""http://www.w3.org/TR/REC-html40/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc on Sat Sep 20 09:42:55 CDT 2003 -->
<TITLE>
Genotype (JGAP 1.0 API)
</TITLE>
<LINK REL ="stylesheet" TYPE="text/css" HREF="../../stylesheet.css" TITLE="Style">
</HEAD>
<SCRIPT>
function asd()
{
parent.document.title="Genotype (JGAP 1.0 API)";
}
</SCRIPT>
<BODY BGCOLOR="white" onload="asd();">

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_top"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgap/FitnessFunction.html"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Genotype.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.jgap</FONT>
<BR>
Class  Genotype</H2>
<PRE>
java.lang.Object
  |
  +--<B>org.jgap.Genotype</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.io.Serializable</DD>
</DL>
<HR>
<DL>
<DT>public class <B>Genotype</B><DT>extends java.lang.Object<DT>implements java.io.Serializable</DL>

<P>
Genotypes are fixed-length populations of chromosomes. As an instance of
 a Genotype is evolved, all of its Chromosomes are also evolved. A Genotype
 may be constructed normally, whereby an array of Chromosomes must be
 provided, or the static randomInitialGenotype() method can be used to
 generate a Genotype with a randomized Chromosome population.
<P>

<P>
<DL>
<DT><B>Since:</B><DD>1.0</DD>
</DD>
<DT><B>Author:</B><DD>Neil Rotstan</DD>
</DD>
<DT><B>See Also:</B><DD><A HREF="../../serialized-form.html" TARGET="org.jgap.Genotype">Serialized Form</A></DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->


<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Field Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgap/Configuration.html">Configuration</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#m_activeConfiguration">m_activeConfiguration</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The current active Configuration instance.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgap/Chromosome.html">Chromosome</A>[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#m_chromosomes">m_chromosomes</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The array of Chromosomes that makeup thie Genotype's population.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#m_workingPool">m_workingPool</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The working pool of Chromosomes, which is where Chromosomes that are
 to be candidates for the next natural selection process are deposited.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#Genotype(org.jgap.Configuration, org.jgap.Chromosome[])">Genotype</A></B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration,
         <A HREF="../../org/jgap/Chromosome.html">Chromosome</A>[]&nbsp;a_initialChromosomes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a new Genotype instance with the given array of
 Chromosomes and the given active Configuration instance.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=2><FONT SIZE="+2">
<B>Method Summary</B></FONT></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Genotype against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#evolve()">evolve</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Evolve the population of Chromosomes within this Genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgap/Chromosome.html">Chromosome</A>[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#getChromosomes()">getChromosomes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the array of Chromosomes that make up the population of this
 Genotype instance.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgap/Chromosome.html">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#getFittestChromosome()">getFittestChromosome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the Chromosome in the population with the highest fitness
 value.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../org/jgap/Genotype.html">Genotype</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#randomInitialGenotype(org.jgap.Configuration)">randomInitialGenotype</A></B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convenience method that returns a newly constructed Genotype
 instance configured according to the given Configuration instance.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#setActiveConfiguration(org.jgap.Configuration)">setActiveConfiguration</A></B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the active Configuration object on this Genotype and its
 member Chromosomes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgap/Genotype.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Return a string representation of this Genotype instance,
 useful for dispaly purposes.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TD><B>Methods inherited from class java.lang.Object</B></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Field Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="m_activeConfiguration"><!-- --></A><H3>
m_activeConfiguration</H3>
<PRE>
protected transient <A HREF="../../org/jgap/Configuration.html">Configuration</A> <B>m_activeConfiguration</B></PRE>
<DL>
<DD>The current active Configuration instance.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_chromosomes"><!-- --></A><H3>
m_chromosomes</H3>
<PRE>
protected <A HREF="../../org/jgap/Chromosome.html">Chromosome</A>[] <B>m_chromosomes</B></PRE>
<DL>
<DD>The array of Chromosomes that makeup thie Genotype's population.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_workingPool"><!-- --></A><H3>
m_workingPool</H3>
<PRE>
protected transient java.util.List <B>m_workingPool</B></PRE>
<DL>
<DD>The working pool of Chromosomes, which is where Chromosomes that are
 to be candidates for the next natural selection process are deposited.
 This list is passed to each of the genetic operators as they are
 invoked during each phase of evolution so that they can add the
 Chromosomes they operated upon to it, and then it is eventually passed
 to the NaturalSelector so that it can choose which Chromosomes will
 go on to the next generation and which will be discarded. It is wiped
 clean after each cycle of evolution.
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="Genotype(org.jgap.Configuration, org.jgap.Chromosome[])"><!-- --></A><H3>
Genotype</H3>
<PRE>
public <B>Genotype</B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration,
                <A HREF="../../org/jgap/Chromosome.html">Chromosome</A>[]&nbsp;a_initialChromosomes)
         throws <A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></PRE>
<DL>
<DD>Constructs a new Genotype instance with the given array of
 Chromosomes and the given active Configuration instance. Note
 that the Configuration object must be in a valid state
 when this method is invoked, or a InvalidconfigurationException
 will be thrown.
<P>

<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if either the given Configuration object
         or the array of Chromosomes is null, or if any of the Genes
         in the array of Chromosomes is null.
<DD><CODE><A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></CODE> - if the given Configuration object
         is in an invalid state.</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" WIDTH="100%">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TD COLSPAN=1><FONT SIZE="+2">
<B>Method Detail</B></FONT></TD>
</TR>
</TABLE>

<A NAME="setActiveConfiguration(org.jgap.Configuration)"><!-- --></A><H3>
setActiveConfiguration</H3>
<PRE>
public void <B>setActiveConfiguration</B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)
                            throws <A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></PRE>
<DL>
<DD>Sets the active Configuration object on this Genotype and its
 member Chromosomes. This method should be invoked immediately following
 deserialization of this Genotype. If an active Configuration has already
 been set on this Genotype, then this method will do nothing.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current active Configuration object
                              that is to be referenced internally by
                              this Genotype and its member Chromosome
                              instances.
<DT><B>Throws:</B>
<DD><CODE><A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></CODE> - if the Configuration object is
         null or cannot be locked because it is in an invalid or
         incomplete state.</DL>
</DD>
</DL>
<HR>

<A NAME="getChromosomes()"><!-- --></A><H3>
getChromosomes</H3>
<PRE>
public <A HREF="../../org/jgap/Chromosome.html">Chromosome</A>[] <B>getChromosomes</B>()</PRE>
<DL>
<DD>Retrieves the array of Chromosomes that make up the population of this
 Genotype instance.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The population of Chromosomes.</DL>
</DD>
</DL>
<HR>

<A NAME="getFittestChromosome()"><!-- --></A><H3>
getFittestChromosome</H3>
<PRE>
public <A HREF="../../org/jgap/Chromosome.html">Chromosome</A> <B>getFittestChromosome</B>()</PRE>
<DL>
<DD>Retrieves the Chromosome in the population with the highest fitness
 value.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The Chromosome with the highest fitness value, or null if
         there are no chromosomes in this Genotype.</DL>
</DD>
</DL>
<HR>

<A NAME="evolve()"><!-- --></A><H3>
evolve</H3>
<PRE>
public void <B>evolve</B>()</PRE>
<DL>
<DD>Evolve the population of Chromosomes within this Genotype. This will
 execute all of the genetic operators added to the present active
 Configuration and then invoke the natural selector to choose which
 chromosomes will be included in the next generation population. Note
 that the population size always remains constant.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD>Return a string representation of this Genotype instance,
 useful for dispaly purposes.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A string representation of this Genotype instance.</DL>
</DD>
</DL>
<HR>

<A NAME="randomInitialGenotype(org.jgap.Configuration)"><!-- --></A><H3>
randomInitialGenotype</H3>
<PRE>
public static <A HREF="../../org/jgap/Genotype.html">Genotype</A> <B>randomInitialGenotype</B>(<A HREF="../../org/jgap/Configuration.html">Configuration</A>&nbsp;a_activeConfiguration)
                                      throws <A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></PRE>
<DL>
<DD>Convenience method that returns a newly constructed Genotype
 instance configured according to the given Configuration instance.
 The population of Chromosomes will created according to the setup of
 the sample Chromosome in the Configuration object, but the gene values
 (alleles) will be set to random legal values.
 <p>
 Note that the given Configuration instance must be in a valid state
 at the time this method is invoked, or an InvalidConfigurationException
 will be thrown.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A newly constructed Genotype instance.
<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if the given Configuration object is
         null.
<DD><CODE><A HREF="../../org/jgap/InvalidConfigurationException.html">InvalidConfigurationException</A></CODE> - if the given Configuration
         instance not in a valid state.</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this Genotype against the specified object. The result is true
 if the argument is an instance of the Genotype class, has exactly the
 same number of chromosomes as the given Genotype, and, for each
 Chromosome in this Genotype, there is an equal chromosome in the
 given Genotype. The chromosomes do not need to appear in the same order
 within the populations.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>

<!-- ========== START OF NAVBAR ========== -->
<A NAME="navbar_bottom"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0">
<TR>
<TD COLSPAN=3 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgap/FitnessFunction.html"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html" TARGET="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Genotype.html" TARGET="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;
<SCRIPT>
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
<A HREF="../../allclasses-noframe.html" TARGET=""><B>All Classes</B></A>
</NOSCRIPT>
</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- =========== END OF NAVBAR =========== -->

<HR>

</BODY>
</HTML>
